### ISSUE TITLE
Empty audio file - invalid value encountered in cast  data = data.astype(np.int16)

### ISSUE THREAD

**Issue by zerawww** [SWivid/F5-TTS#204]
Win11
NVDIA GTX 1660ti
torch 2.3.0
cu118
python 3.10

LOG:
Starting app...
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
C:\Users\frapp\anaconda3\envs\F5\lib\site-packages\transformers\models\whisper\generation_whisper.py:496: FutureWarning: The input name `inputs` is deprecated. Please make sure to use `input_features` instead.
  warnings.warn(
You have passed task=transcribe, but also have set `forced_decoder_ids` to [[1, None], [2, 50359]] which creates a conflict. `forced_decoder_ids` will be ignored in favor of task=transcribe.
C:\Users\frapp\anaconda3\envs\F5\lib\site-packages\transformers\models\whisper\modeling_whisper.py:599: UserWarning: 1Torch was not compiled with flash attention. (Triggered internally at ..\aten\src\ATen\native\transformers\cuda\sdp_utils.cpp:455.)
  attn_output = torch.nn.functional.scaled_dot_product_attention(
Passing a tuple of `past_key_values` is deprecated and will be removed in Transformers v4.43.0. You should pass an instance of `EncoderDecoderCache` instead, e.g. `past_key_values=EncoderDecoderCache.from_legacy_cache(past_key_values)`.
The attention mask is not set and cannot be inferred from input because pad token is same as eos token. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
gen_text 0 the audio file is always empty
Building prefix dict from the default dictionary ...
Dumping model to file cache C:\Users\frapp\AppData\Local\Temp\jieba.cache
Loading model cost 0.784 seconds.
Prefix dict has been built successfully.
C:\Users\frapp\anaconda3\envs\F5\lib\site-packages\torch\functional.py:665: UserWarning: ComplexHalf support is experimental and many operators don't support it yet. (Triggered internally at ..\aten\src\ATen\EmptyTensor.cpp:42.)
  return _VF.stft(input, n_fft, hop_length, win_length, window,  # type: ignore[attr-defined]
C:\Users\frapp\anaconda3\envs\F5\lib\site-packages\gradio\processing_utils.py:574: UserWarning: Trying to convert audio automatically from float32 to 16-bit int format.
  warnings.warn(warning.format(data.dtype))
C:\Users\frapp\anaconda3\envs\F5\lib\site-packages\gradio\processing_utils.py:577: RuntimeWarning: invalid value encountered in cast
  data = data.astype(np.int16)

**Comment by SWivid** [comment]
not sure if it's because of 1660ti not supporting well fp16 or something.
maybe try comment out
https://github.com/SWivid/F5-TTS/blob/a79199e54d99eea660e8a1480788a49579cc5ba2/model/utils.py#L595-L596

**Comment by zerawww** [comment]
That was it, Works without issue now.
Thanks!

**Comment by NinjaPerson24119** [comment]
Thanks. This fixed empty audio with ROCm on 7900 XTX as well.
